### DISCUSSION TITLE
Unable to Connect to Internet

### DISCUSSION THREAD

**Discussion by camman61898** [Winston-Lu/ESP8266-LED-Shelf#10] [Q&A] [answered]
Hey, I am not sure if this is still active or not, but I went through all of the steps outlined here and I tried to flash my NodeMCU with the firmware after creating the tab called Secrets.h and filled out the proper ssid and password, but I have had no luck getting it to connect to the internet. I also downloaded the littlefs tool and I believe I all of the data folder properly uploaded, but I have no way to know without getting the board also connected to the internet. Do you have any idea what the issue could be? Thank you!

**Comment by camman61898** [comment, marked as answer]
Also, it seems to compile correctly and it uploads to the board just fine, but it does not show up on my router when I check to see if the device has been added.

**Reply by Winston-Lu** [reply]
I'm still providing bugfixes but not currently working on new features. Keep the board plugged into the computer and let me know what it says in the serial monitor. It should keep printing ...... if it's still trying to connect, here might be some issues:
- Incorrect SSID setup. Maybe it is trying  to connect to the 5GHz network when the ESP8266 only can do 2.4GHz? 
- Could be trying to connect with a static IP that is already in use 
  - Connects as 192.168.1.51 if I remember correctly. If another device on the network is using that IP, it will either fallback to your routers DHCP protocol or wont be allowed to connect
- Run out of DCHP addresses (unlikely, but possible if you have lots of smart home devices)
- Bad connection to the router. I believe there is an example sketch for the ESP8266 to show what network devices it can see with its, not great but not bad antenna. 

Otherwise it will print out a bunch of other information after it connects. I believe the network name (SSID) is case sensitive as well. If it connects to your router, it should be good to go to connect to the internet. If it connects but doesn't appear on your router, you can try going to the IP printed in the serial monitor to connect, otherwise this is likely an issue on your router side.

**Reply by camman61898** [reply]
I still have had no luck getting it to connect to the internet. I have it connected to the computer just fine, and the firmware flashes to the board just fine... I think, but for some reason it will not connect to WiFi. I have tried two different NodeMCU's, and I have tried to connect it to both my router that my VPN is hosted on, and my main WiFi 6E Router. Neither of them will show my device online. I have tried also turning off my password on my router and that didn't seem to help either. I got it to connect to the router with a different program, but I have yet to get it to work with this one, and I am really looking forward to getting it to work since I won't have to worry about having to physically plug into the Arduino to update the time like I would have in the original post by DIY Machines. This is what it should look like, correct?
![Arduino](https://user-images.githubusercontent.com/93802067/177903105-01222ac5-ba4f-4116-808d-8eabaa1478ec.PNG)

**Reply by Winston-Lu** [reply]
It looks correct. Check on your router to see if you enabled the 2.4GHz network. You might be using the SSID for the 5 or 6GHz band

**Reply by camman61898** [reply]
I wish that was it, but the router that my VPN is connected to is only a 2.4GHz router and it has no option for 5 GHz, so I am unsure what the issue could be.
![Arduino2](https://user-images.githubusercontent.com/93802067/177903792-63eeb355-fb4b-40e6-8d4c-fffd505084d7.PNG)
This is what the file looks like when I compile and flash it onto the board. Does this look right?

**Reply by Winston-Lu** [reply]
That seems to just be the upload section. In the top right of the Arduino IDE, there should be a button for the serial monitor. 
![image](https://user-images.githubusercontent.com/33874247/178032014-76155a09-0680-49e5-966c-4525935abf42.png)
From there, it should so the debug info. I don't have an ESP8266 on me at the moment, so I wont be able to provide a more detailed screenshot

**Reply by camman61898** [reply]
This is what the Serial Monitor shows.

..
mDNS set up. You can connect to ledshelf.local to connect instead of the IP
UTC Offset: 0
Time updated
Showing lights
Getting webserver started...
HTTP web server initialized
Maximum FPS: 561.8     Milliseconds per frame: 1.78
Maximum FPS: 551.6     Milliseconds per frame: 1.81
Maximum FPS: 427.5     Milliseconds per frame: 2.34
Maximum FPS: 408.0     Milliseconds per frame: 2.45
Maximum FPS: 408.5     Milliseconds per frame: 2.45

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (29):
epc1=0x4020589a epc2=0x00000000 epc3=0x00000000 excvaddr=0x00649680 depc=0x00000000

>>>stack>>>

ctx: cont
sp: 3ffffda0 end: 3fffffc0 offset: 0190
3fffff30:  3fffdad0 3ffe85ca 3ffefe78 40205878  
3fffff40:  3fffdad0 3ffe85ca 3ffefe78 40205f4a  
3fffff50:  6978614d 206d756d 3a535046 38303420  
3fffff60:  007a1200 5f50dcaf 696c6c00 6f636573  
3fffff70:  00000000 20726570 6d617266 32203a65  
3fffff80:  00000000 00000000 00000001 401003ec  
3fffff90:  3fffdad0 00000000 3fff0c38 4010040d  
3fffffa0:  3fffdad0 00000000 3fff0c38 4021a264  
3fffffb0:  feefeffe feefeffe 3ffe85f0 40100ef5  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3460, room 16 
tail 4
chksum 0xcc
load 0x3fff20b8, len 40, room 4 
tail 4
chksum 0xc9
csum 0xc9
v00060310
~ld






Initializing data structures for lighting effects
Loading EEPROM
Done loading from EEPROM. Loaded 69 bytes up to 0x48
Done initializing lighting data
LittleFS contents:
FS File: config.js, size: 364
FS File: css, size: 0
FS File: img, size: 0
FS File: index.html, size: 7314

Setting up Wi-Fi and NTP Connection
..
mDNS set up. You can connect to ledshelf.local to connect instead of the IP
UTC Offset: 0
Time updated
Showing lights
Getting webserver started...
HTTP web server initialized
Maximum FPS: 512.8     Milliseconds per frame: 1.95


So it does seem to be uploading fine, but for some reason, it wont connect to the internet...

**Reply by Winston-Lu** [reply]
On your computer, type `ipconfig` or `ifconfig` in command prompt/terminal for windows or mac respectively. Look for the default gateway on your wireless network card. It should be something like 192.168.#.# or 10.0.#.#. Make sure that is correctly setup like so in NTPTime.cpp here. The `ip` you can try changing around. `gateway` should be the default gateway you found above. I'll look into the code further on my side as well
```
//IP config
IPAddress ip(192,168,1,52);     //Device IP
IPAddress gateway(192,168,1,1); //IP of router
IPAddress subnet(255,255,255,0);
IPAddress primaryDNS(8,8,8,8);
IPAddress secondaryDNS(8,8,4,4);
```

**Reply by camman61898** [reply]
I will try that tonight, the IP address of my router is 192.168.8.1. so that could be it if I understood correctly

**Reply by camman61898** [reply]
YOU GENIUS!!!! It worked!
![IT WORKED](https://user-images.githubusercontent.com/93802067/178078479-bae4d2e5-13b8-495e-984a-62839eba9c32.PNG)

**Reply by camman61898** [reply]
Well... I think, I am unable to run anything in the command line at the bottom of the file. When I click submit, it doesn't do anything. Does that mean that there was an error with the littlefs upload? Or does the board have to be connected to more than just the USB cable to my computer?

**Reply by camman61898** [reply]
Also, does this project require a resistor similar to the original Arduino project?

**Reply by Winston-Lu** [reply]
> Well... I think, I am unable to run anything in the command line at the bottom of the file. When I click submit, it doesn't do anything.

Check the serial monitor to see if there are any outputs. I don't see anything pop up on that website (RGB color pickers), so that usually means the ESP is still trying to send data to the website from the backend and something may have went wrong. Send me the serial monitor data and press F12 on the website and paste what it says in the `Console` tab. I think the file upload went fine, might be something wrong with the code. The board should just need a power source.

> Also, does this project require a resistor similar to the original Arduino project?

If you are talking about the resistor between the ESP8266 and the Din pin on the light strip, I would recommend it but it is not necessary as it is more of a precaution/failsafe component.

**Reply by camman61898** [reply]
![Console Tab](https://user-images.githubusercontent.com/93802067/178088521-66e18da9-85e7-40bd-8832-e10ee20771b6.PNG)
There does seem to be an issue there...


Below is the Serial Monitor
mDNS set up. You can connect to ledshelf.local to connect instead of the IP
UTC Offset: 0
Time updated
Showing lights
Getting webserver started...
HTTP web server initialized
Maximum FPS: 464.7     Milliseconds per frame: 2.15
Maximum FPS: 460.8     Milliseconds per frame: 2.17
Maximum FPS: 480.3     Milliseconds per frame: 2.08
Maximum FPS: 458.3     Milliseconds per frame: 2.18
Maximum FPS: 462.1     Milliseconds per frame: 2.16

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (29):
epc1=0x4020589a epc2=0x00000000 epc3=0x00000000 excvaddr=0x00649680 depc=0x00000000

>>>stack>>>

ctx: cont
sp: 3ffffda0 end: 3fffffc0 offset: 0190
3fffff30:  3fffdad0 3ffe85ca 3ffefe78 40205878  
3fffff40:  3fffdad0 3ffe85ca 3ffefe78 40205f4a  
3fffff50:  6978614d 206d756d 3a535046 32363420  
3fffff60:  007a1200 2f4b748a 696c6c00 6f636573  
3fffff70:  00000000 20726570 6d617266 32203a65  
3fffff80:  00000000 00000000 00000001 401003ec  
3fffff90:  3fffdad0 00000000 3fff0c38 4010040d  
3fffffa0:  3fffdad0 00000000 3fff0c38 4021a264  
3fffffb0:  feefeffe feefeffe 3ffe85f0 40100ef5  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 3460, room 16 
tail 4
chksum 0xcc
load 0x3fff20b8, len 40, room 4 
tail 4
chksum 0xc9
csum 0xc9
v00060310
~ld






Initializing data structures for lighting effects
Loading EEPROM
Done loading from EEPROM. Loaded 69 bytes up to 0x48
Done initializing lighting data
LittleFS contents:
FS File: config.js, size: 364
FS File: css, size: 0
FS File: img, size: 0
FS File: index.html, size: 7314

Setting up Wi-Fi and NTP Connection
..
mDNS set up. You can connect to ledshelf.local to connect instead of the IP
UTC Offset: 0
Time updated
Showing lights
Getting webserver started...
HTTP web server initialized
Maximum FPS: 491.2     Milliseconds per frame: 2.04
Maximum FPS: 491.6     Milliseconds per frame: 2.03
Maximum FPS: 485.9     Milliseconds per frame: 2.06
Maximum FPS: 522.5     Milliseconds per frame: 1.91


Below are my files and folders.
![LED_CLOCK Files](https://user-images.githubusercontent.com/93802067/178088668-42be6467-adee-4433-b067-c03e58e4c69a.PNG)
![Data Files](https://user-images.githubusercontent.com/93802067/178088670-fc77c687-6918-4098-aaa4-10eb54867adc.PNG)

**Reply by Winston-Lu** [reply]
Looks like it cant find the color picker. Try uploading it again using this tool here:
![image](https://user-images.githubusercontent.com/33874247/178089011-3e78e069-67fa-47e1-b329-6bdda80ad21a.png)
Using these settings (I only changed the Flash Size to 4MB, with 1MB FS or greater)
![image](https://user-images.githubusercontent.com/33874247/178089057-cf0d1848-6a7a-4895-bcbe-47b9d4c98aca.png)
Then upload the sketch, then use the LittleFS tool to upload the website. [Exception 29](https://arduino-esp8266.readthedocs.io/en/latest/exception_causes.html) is trying to store in a prohibited address, so it might be something wrong with the EEPROM or with the flash size configuration

**Reply by camman61898** [reply]
I had changed the flash size to that size, but somehow it got changed, I probably changed it when I couldnt get it to connect to the internet. That fixed it and I now have all of the color selectors and it all seems to be working! Now to wire it up and see if it actually works! I am going to do what another person on here did and that is to use the default wiring diagram from the original post. So I will have to use the reorder line that you commented in the previous discussion about that. The DIY Machines version has the Clock LED's wired separately from the spotlights. I will have to change some of the code, correct?

**Reply by Winston-Lu** [reply]
Yes, the README on the main page documents which lines to edit, but for the sake of not reading the very long documentation, all you need to change is in `Config.h`, just uncomment `#define SPOTLIGHTPIN 4` and set it to the respective pin and it should be good to go

**Reply by camman61898** [reply]
You are such a good help, I really appreciate it! I also noticed in your other discussion that you don't want tips, but if that has changed, let me know and I will gladly send a little extra cash your way, I really appreciate it!

**Reply by camman61898** [reply]
I just got it hooked up and plugged it in for the first time, it is AWESOME!! The only thing I have noticed is that the autobrightness is backwards.. When it gets dark, the clock gets brighter and vice versa. Would I change the if(autobrightness) line? I am not sure which line I should change though.

**Reply by Winston-Lu** [reply]
Yeah, I don't really use auto-brightness so its not tested very much. You would be correct, it would be the following lines in `Lightning.cpp`:
```c++
if(autobrightness){
    double val = (analogRead(LIGHT_SENSOR)/1024.0)*2; //Between 0-2
    segmentBrightness =    min(max((int)( 40.0 * val*val ), 5),150); //Clamp between 5 and 150, median 40
    backgroundBrightness = min(max((int)( 20.0 * val*val ), 2),100); //Clamp between 2 and 100, median 20
    spotlightBrightness =  min(max((int)(160.0 * val*val ),30),255); //Clamp between 30 and 255, median 160
  }
```
If you find a working one you're happy with, you can submit a pull request or just post it here and I'll make the changes. I've played around with the values a bit, but couldn't find one I was happy with. For a quick fix, you can change the `double val` line to this to invert it:
```c++
double val = 2 - (analogRead(LIGHT_SENSOR)/1024.0)*2; //Between 0-2
```

**Reply by camman61898** [reply]
That worked well for me! The code I ended up using is below. The only thing that I notice is that the lights kinda flicker since they change instantly with any value they receive from the photocell. Is there anyway to add a line of code that would give it like a +/- 5 or so buffer to it so it wouldn't change value unless the received value was a multiple of 5?

  if(autobrightness){
    double val = 2 - (analogRead(LIGHT_SENSOR)/1024.0)*2; //Between 0-2
    segmentBrightness =    min(max((int)( 100.0 * val*val ), 20),255); //Clamp between 20 and 255, median 100
    backgroundBrightness = min(max((int)( 20.0 * val*val ), 2),100); //Clamp between 2 and 100, median 20
    spotlightBrightness =  min(max((int)(160.0 * val*val ),20),255); //Clamp between 20 and 255, median 160
  }

**Reply by Winston-Lu** [reply]
I'll push a commit to make it smoother, I think I have another idea in mind but implementing it would be a bit in the tricker side if I cant test it so I won't make any promises here yet. I'll edit this comment when its done.

**Reply by camman61898** [reply]
Okay awesome! Could you add a reply to this thread so I get an email update when you update it? Thank you!

**Reply by Winston-Lu** [reply]
Alright, new commit is out. Test it out and let me know if there's any issue. What it should do now is get an average light value every second, and average the past 5 samples by default. If the lighting value changes, it should transition to the new light value each frame. You can change these values in `Config.h`, where you can change the delay and samples here
```c++
//autobrightness stuff
#define AUTOBRIGHTNESS_DELAY 1 //seconds per brightness update. Supports decimal values
#define AUTOBRIGHTNESS_SAMPLES 5 //number of samples to average. Must be an integer
```
